packages/flutter_tools/templates/plugin/test/projectName_test.dart.tmpl

```diff
@@ -1,23 +1,27 @@
-import 'package:flutter/services.dart';
 import 'package:flutter_test/flutter_test.dart';
 import 'package:{{projectName}}/{{projectName}}.dart';
+import 'package:{{projectName}}/{{projectName}}_platform_interface.dart';
+import 'package:plugin_platform_interface/plugin_platform_interface.dart';
 
-void main() {
-  const MethodChannel channel = MethodChannel('{{projectName}}');
+class Mock{{pluginDartClass}}Platform 
+    with MockPlatformInterfaceMixin
+    implements {{pluginDartClass}}Platform {
 
-  TestWidgetsFlutterBinding.ensureInitialized();
+  @override
+  Future<String?> get platformVersion => Future.value('42');
+}
 
+void main() {
+  late {{pluginDartClass}} {{pluginClassLowerCamelCase}};
+  late Mock{{pluginDartClass}}Platform fakePlatform;
+  
   setUp(() {
-    channel.setMockMethodCallHandler((MethodCall methodCall) async {
```

Comment:
All of the code you've removed from this file should still exist, just as a file called `projectName_method_channel_test.dart` (written against the platform interface though).